exercises/hangman/HangmanTest.cs

```diff
@@ -1,135 +1,342 @@
+        var player2 = scheduler.CreateObserver<HangmanState>();
+        var game = new Hangman("hello");
 
-        game.Guess('e');
+        var player1 = game.StateObservable;
+        Ready(player1);
 
-        Assert.Equal(HangmanStatus.Busy, lastState.Status);
-        Assert.Equal(8, lastState.RemainingGuesses);
-        Assert.Equal("_e___", lastState.MaskedWord);
+        scheduler.Schedule(TimeSpan.FromTicks(100), () => game.GuessObserver.OnNext('e'));
+        scheduler.Schedule(TimeSpan.FromTicks(200), () => game.GuessObserver.OnNext('l'));
+        scheduler.Schedule(TimeSpan.FromTicks(150), () => game.StateObservable.Subscribe(player2));
 
-        game.Guess('l');
+        var expected = new[]
+        {
+            OnNext<HangmanState>(150, hangmanState => hangmanState.RemainingGuesses == 9 && hangmanState.MaskedWord == "_e___"),
+            OnNext<HangmanState>(200, hangmanState => hangmanState.RemainingGuesses == 9 && hangmanState.MaskedWord == "_ell_")
+        };
+
+        // Act
+        // +--e--l->
+        // +a-b--c->
+        // ...+b-c->
+        scheduler.Start();
+
+        // Assert
+        ReactiveAssert.AreElementsEqual(expected, player2.Messages);
+    }
 
-        Assert.Equal(HangmanStatus.Busy, lastState.Status);
-        Assert.Equal(8, lastState.RemainingGuesses);
-        Assert.Equal("_ell_", lastState.MaskedWord);
+    private IDisposable Ready(IObservable<HangmanState> player)
+    {
+        return player.Subscribe(x => { });
+    }
 
-        game.Guess('o');
+    // Expert mode on>
+    [Fact(Skip = "Remove to run test")]
+    public void Multiple_player_sees_the_same_game_already_started()
+    {
+        // Arrange
+        TestScheduler scheduler = new TestScheduler();
+        var player2 = scheduler.CreateObserver<HangmanState>();
+        var player3 = scheduler.CreateObserver<HangmanState>();
+        var game = new Hangman("hello");
+
+        var player1 = game.StateObservable;
+        Ready(player1);
+
+        scheduler.Schedule(TimeSpan.FromTicks(100), () => game.GuessObserver.OnNext('e'));
+        scheduler.Schedule(TimeSpan.FromTicks(200), () => game.GuessObserver.OnNext('l'));
+        scheduler.Schedule(TimeSpan.FromTicks(150), () =>
+            {
+                game.StateObservable.Subscribe(player2);
+                game.StateObservable.Subscribe(player3);
+            });
+
+        var expected = new[]
+        {
+            OnNext<HangmanState>(150, hangmanState => hangmanState.RemainingGuesses == 9 && hangmanState.MaskedWord == "_e___"),
+            OnNext<HangmanState>(200, hangmanState => hangmanState.RemainingGuesses == 9 && hangmanState.MaskedWord == "_ell_"),
+        };
+
+        // Act
+        // +--e--l->
+        // +a-b--c->
+        // ...+b-c->
+        // ...+b-c->
+        scheduler.Start();
+
+        // Assert
+        // hint: http://reactivex.io/documentation/subject.html have you noticed that Subject are both IObservable, and IObserver?
```

Comment:
I think this gives away a bit too much information. As this is one of the hardest exercises, we can expect the students to figure this out themselves I hope.